### ISSUE TITLE
Plumb stdout and stderr through CellService

### ISSUE THREAD

**Issue by krisnova** [aurae-runtime/aurae#173]
I believe we will need to develop a logging cache that serves as an in-memory caching layer (Note: in the future it will need to also persist to disk!) for both stdout and stderr streams from an executable within a cell.

We will need to be able to hook into a stream at any moment and have some basic guarantees about the data and how we retrieve it.

**Comment by dmah42** [comment]
i think this fits nicely as something that's a member of the `Executable` struct we now have.

**Comment by dmah42** [comment]
i had a go at this, and with the introduction of `unshare` it's become a bit tricky.

tl;dr: unshare allows us to set up pipes for stdout and stderr, which is great (the default "spawn" on std::process inherits the parent's pipes).  however, any attempt to read these pipes is stymied by unshare not implementing any of the `io::Read` trait _publicly_ for the `PipeReader`.

i was hoping to create a thread that would read stdio/stderr into buffers using the `io::Read` trait implementation until the command exits or is killed (at which point we'd want to join the thread, obviously).

best option i can see is to convert the pipes to `RawFd` and then (unsafe) create `File` out of each for buffered reading.  it should work, but it's not great.

**Comment by krisnova** [comment]
I believe this is the next task for us (along with other general code improvements)

**Comment by dmah42** [comment]
I'm happy to take another crack if noone else gets to it.
